### PR TITLE
Added default mode prompt function

### PR DESCRIPTION
**Pull request by cprieto**
## Description

Now fish_mode_prompt calls fish_default_mode_prompt

Fixes issue #3641

## TODOs:

- [x] Changes to fish usage are reflected in user documenation/manpages.
- [ ] Tests have been added for regressions fixed

### CODE REVIEW

File: doc_src/index.hdr.in [hunk 1]
```diff
@@ -1050,7 +1050,7 @@ function fish_user_key_bindings
 end
 \endfish
 
-When in vi-mode, the <a href="fish_mode_prompt.html">`fish_mode_prompt`</a> function will display a mode indicator to the left of the prompt. The `fish_vi_cursor` function will be used to change the cursor's shape depending on the mode in supported terminals. To disable this feature, override it with an empty function.
+When in vi-mode, the <a href="fish_mode_prompt.html">`fish_mode_prompt`</a> function will display a mode indicator to the left of the prompt. The `fish_vi_cursor` function will be used to change the cursor's shape depending on the mode in supported terminals. To disable this feature, override it with an empty function. Currently `fish_mode_prompt` calls `fish_default_mode_prompt`, this means you could use `fish_default_mode_prompt` to display the cursor or default cursor somewhere else.
```
Comment (faho) [inline doc_src/index.hdr.in]: Drop the "currently". The documentation describes how things are, unless we have explicit plans to change it. In this case we don't.

We want to make a bunch of things clear here:

- fish_mode_prompt can be changed to an empty function to disable the feature

- fish_mode_prompt can be changed to something else to display it differently

- fish_default_mode_prompt can be used elsewhere to display the mode in other places - not at the beginning of the line

This also has nothing to do with a "cursor", that would be `fish_vi_cursor`, which is a different feature.

So how about this:

"To display the mode elsewhere (like in your right prompt), use the output of the `fish_default_mode_prompt` function"?

File: share/functions/fish_default_mode_prompt.fish [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                set_color --bold --background green white
+                echo '[R]'
+            case visual
+                set_color --bold --background magenta white
+                echo '[V]'
+        end
+        set_color normal
+        echo -n ' '
+    end
+end
```
Comment (faho) [inline share/functions/fish_default_mode_prompt.fish]: Missing newline. Not the end of the world (that'll be if someone invents mushroom-flavored ice cream with chunks).

File: share/functions/fish_mode_prompt.fish [hunk 3]
```diff
@@ -1,23 +1,5 @@
...
-                set_color --bold --background green white
-                echo '[R]'
-            case visual
-                set_color --bold --background magenta white
-                echo '[V]'
-        end
-        set_color normal
-        echo -n ' '
-    end
+    # by default it calls the default mode prompt
```
Comment (faho) [inline share/functions/fish_mode_prompt.fish]: We can see that it calls the default mode prompt, the important question is "Why".

And the answer is that we do it so the indicator can be reused.

**Comment by faho** [comment]
One more thing: We prefer present tense for commit messages - "Add" instead of "Added".

This not only emphasizes the _commit_ instead of the author ("I added when I wrote this" vs "This adds when you apply it"), but also helps keep the subject line shorter ("added" is two characters longer than "add").

This doesn't have to change here, it's just something to keep in mind for future commits.

**Comment by cprieto** [comment]
Done! Thanks @faho 😄

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**